crates/fee/src/lib.rs

```diff
@@ -494,11 +466,16 @@ impl<T: Config> Pallet<T> {
             .ok_or(Error::<T>::TryIntoIntError)?)
     }
 
-    pub fn distribute_global_pool<Reward: reward::Rewards<T::AccountId>>(account_id: &T::AccountId) -> DispatchResult {
-        let reward_as_inner = Reward::withdraw_reward(RewardPool::Global, account_id)?;
+    pub fn distribute_from_reward_pool<
+        Rewards: reward::Rewards<T::AccountId, SignedFixedPoint = SignedFixedPoint<T>>,
+        Staking: staking::Staking<T::AccountId, SignedFixedPoint = SignedFixedPoint<T>>,
+    >(
+        account_id: &T::AccountId,
+    ) -> DispatchResult {
+        let reward_as_inner = Rewards::withdraw_reward(account_id)?;
         let reward_as_fixed =
-            Reward::SignedFixedPoint::checked_from_integer(reward_as_inner).ok_or(Error::<T>::TryIntoIntError)?;
-        Reward::distribute(RewardPool::Local(account_id.clone()), reward_as_fixed)?;
+            Rewards::SignedFixedPoint::checked_from_integer(reward_as_inner).ok_or(Error::<T>::TryIntoIntError)?;
+        Staking::distribute_reward(account_id, reward_as_fixed)?;
         Ok(())
```

Comment:
Here we take rewards earned via the Vault's SLA and transfer to the staking pool - which may have nominated collateral.

![image](https://user-images.githubusercontent.com/12613821/124585467-38b27680-de4d-11eb-8d04-1e969a399b18.png)

Reply:
This is a great figure, can you add that to the spec as well?